Ashvds/doco udate fixes #55

ashvds opened the pull request:
Creates the first draft of a user guide.
Also fixes help text which wasn't updated when `select` was replaced with `start`.

Requires the github site to point to the userguide file.

ashvds committed:
docco day 1

ashvds committed:
userguide day 2 updates

ashvds committed:
simple example complete

ashvds committed:
first version of userguide fixes #55

ashvds committed:
missing copy for when to use

ashvds committed:
more verbage around why to use it + links in readme

ashvds committed:
some rewording in userguide

timbod7 commented on docs/user-guide.md:
```diff
@@ -0,0 +1,34 @@
+
+## Why use hx-deploy-tool
+
+You've decided that you want finer control over your environment than running it in an aplication or function PAAS, and you need to manage your deployments.
```
typo

ashvds replied:
fixed

timbod7 commented on docs/user-guide.md:
```diff
@@ -0,0 +1,34 @@
+You've decided that you want finer control over your environment than running it in an aplication or function PAAS, and you need to manage your deployments.
+You have a single application or service, and when deploying, need to orcestrate various actions, such as deploying the containers to a single or multiple servers, setting up reverse proxies, managing associating certificates, making sure all the containers are the same version of the application, making sure the logs get ingested etc. etc.
+
+hx-deploy-tool provides a reliable way to deploy all those pieces, and ensure state and version is managed at the same time. It encapsulates all the steps of a deployment, without the need to write script after script. Simply create a docker-compose file, and a couple of moustache templates, and suddently with hx-deploy tool, your deployments are consistent, repeatable, reproducable and highly co ...
```
typo: suddently

ashvds replied:
fixed

timbod7 commented on docs/user-guide.md:
```diff
@@ -0,0 +1,34 @@
+
+hx-deploy-tool provides a reliable way to deploy all those pieces, and ensure state and version is managed at the same time. It encapsulates all the steps of a deployment, without the need to write script after script. Simply create a docker-compose file, and a couple of moustache templates, and suddently with hx-deploy tool, your deployments are consistent, repeatable, reproducable and highly co ...
+
+There is a gap between running software on a single server, and having large container clusters and using various tools for container orhestration. This tool plugs that gap, and can provide consistency and reliability in deployments until you have grown to a place where you need what container orchestration tools provide.
```
typo: orhestration

ashvds replied:
fixed

timbod7 commented on docs/userguide/minimum-requirements.md:
```diff
@@ -0,0 +1,30 @@
+
+## Minimum Requirements - no proxy sidecar
+
+At a minimum, you will need the following (with example files included in the release):
```
I think it's confusing that you list optional requirements in the "minimum requirements section".

ashvds replied:
removed optional lines

timbod7 commented on docs/userguide/minimum-requirements.md:
```diff
@@ -0,0 +1,30 @@
+
+- Server where you will be deploying your release
+- Configuration file (examples included in the release)
+- Release archive - A versioned .zip file Stored in S3 containing:
```
It doesn't need to be S3. I'd link to details of the contents of the zip file, and  write something something like:

- Release store - keeps all published [release zip archives](... link to details of the zip structure...). The release store may either be a path in the local file system or a path in an s3 bucket.

ashvds replied:
updated with your verbage

timbod7 commented on docs/userguide/minimum-requirements.md:
```diff
@@ -0,0 +1,30 @@
+  - **_Optional_** Deployment Scripts
+  - **_Optional_** Binary to be deployed (the deploy command/script can source the binary)
+- **_Optional_** AWS Credentials: see the [official AWS credential page](https://aws.amazon.com/blogs/security/a-new-and-standardized-way-to-manage-credentials-in-the-aws-sdks/)
+- S3 bucket(s) with necessary permissions for r/w - release store and optional state store
```
As per the above, S3 is not necessarily required.

ashvds replied:
added `or directory`

timbod7 commented on docs/userguide/minimum-requirements.md:
```diff
@@ -0,0 +1,30 @@
+  - **_Optional_** Binary to be deployed (the deploy command/script can source the binary)
+- **_Optional_** AWS Credentials: see the [official AWS credential page](https://aws.amazon.com/blogs/security/a-new-and-standardized-way-to-manage-credentials-in-the-aws-sdks/)
+- S3 bucket(s) with necessary permissions for r/w - release store and optional state store
+- Folders with necessary permissions for writing logs
```
I'd write something like:

- The following working directories, writable by the user(s) who will run hx-deploy-tool.

```
/opt/releases     # where releases get unpacked for deployment
/opt/etc/deployment  # where deployment contexts get written
/opt/var/log/    # where the log file gets written
```

Note that the actual directories used can be overwritten in the configuration file.

ashvds replied:
added with minor changes

timbod7 commented on docs/userguide/proxy-example.md:
```diff
@@ -0,0 +1,206 @@
+
+## 1.2. Prepare your environment
+
+In the latest release package, you will find several usefull files to help you get started:
```
typo

ashvds replied:
useful catch

timbod7 commented:
As discussed today we are going to iterate on this stuff. But it's a great start, so lets commit and push forward.

ashvds commented:
Addressed all comments in this commit https://github.com/helix-collective/hx-deploy-tool/commit/8a14ffc59e8e7c8bfdfb3bf05bfde3ce00e6f07c
Moving the updates to gh-pages branch as discussed.
Closing the pull request, which will be re-created from the other branch when ready
